src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtils.java

```diff
@@ -420,4 +451,88 @@ public static boolean isCorrectJavadocPosition(DetailAST blockComment) {
                 || BlockCommentPosition.isOnEnumConstant(blockComment)
                 || BlockCommentPosition.isOnAnnotationDef(blockComment);
     }
+
+    /**
+     * Checks if the supplied {@code DetailNode} argument is not a tight HTML tag.
+     * @param node the node to check
+     * @return true if the javadoc node is not a tight HTML tag; false otherwise
+     */
+    public static boolean isNonTightHtmlTag(DetailNode node) {
+        final boolean isLooseHtmlTag;
+        isLooseHtmlTag = node.getText().endsWith("TAG_START")
+                && node.getParent().getType() != getHtmlTagParentTypeWhenClosed(node);
```

Comment:
@ps-sp , and @Vladlis , both of you are not sure in performance problems. Did you try ?
Performance is minor problem in comparison to unreliable implementation.

When I see code like:
```
NESTED_HTML_CHILD_PARENT_MAP
                .put(JavadocTokenTypes.P_TAG_START, JavadocTokenTypes.PARAGRAPH);
```
and
```
                    case JavadocTokenTypes.P_TAG_START:
                        isNested = toVisitParentType == JavadocTokenTypes.PARAGRAPH;
                        break;
```
Are you sure that you referenced here all tags ? ... how to not forget to add new known tag to such code ? 

it is big sign that rules of javadoc structure become in two places: in grammar and in java code. 
It will never work well. It is error prone. Even now it looks simple, it will be disaster to support. Example is current paring of javadoc by regexps + java code ... it was simple to implement, is more simple to read, work on some simple cases ... BUT it NEVER worked and support of them terrible.

@Vladlis , please find me online/cell for a call, to discuss this.